app/analytics.go

```diff
@@ -244,7 +247,15 @@ func (a *App) GetAnalytics(name string, teamId string) (model.AnalyticsRows, *mo
 		commandChan := make(chan store.StoreResult, 1)
 		go func() {
 			c, err := a.Srv().Store.Command().AnalyticsCommandCount(teamId)
-			commandChan <- store.StoreResult{Data: c, Err: err}
+			var appErr *model.AppError
+			if err != nil {
+				switch {
```

Comment:
For the store methods which don't actually return an AppError, there is no need to do check for this. If it just returns a normal error, we can directly create an AppError and return it / or pass to the channel as done here.

This comment is also applicable to all the other methods in the app layer. From what I see, only Save and Update return an AppError (Save would also return an InvalidInput from my comment below). And Get and GetByTrigger return a NotFound. So only in those cases, we need a switch-case. For the others, we can just directly return.

Reply:
Hi @agnivade , first of all thank you for the really thorough and thoughtful review! Truly appreciate it :slightly_smiling_face: 
1. I think I can change this switch-case on other part like `app/team.go` or `app/user.go` but on this particular case in `app/analytics.go` not using a switch-case got me an error panic of `invalid memory address or nil pointer dereference` from the `TestGetAnalyticsOld` test so that's why I used switch-case on this one. What do you think?

Reply:
Hi @joshuabezaleel - so sorry for the delay here, I somehow completely missed your notification. Regarding this issue, now the `StoreResult` has a `NErr` field added for the purposes of this migration. So you need not check for the error inside the goroutine, but just assign the error to `NErr` than `Err` and check for the value in the caller.

Reply:
No worries at all! Thank you for the pointing it out, @agnivade ! Sorry for missing and not noticing that there is a new field of `NErr`.